### DISCUSSION TITLE
HKSV recording triggered by dummy switch motion with ONVIF camera

### DISCUSSION THREAD

**Discussion by jgbaum** [koush/scrypted#1143] [Q&A]
I've got my Foscam Floodlight set up through the RTSP plugin and motion detection / HKSV recording working using the dummy switch with SMTP after following these instructions:

* https://github.com/koush/scrypted/wiki/HomeKit-Secure-Video-Setup
* https://docs.scrypted.app/dummy-detection.html

I'm curious why this doesn't work with ONVIF though.  Using the same camera, I've configured it through ONVIF and linked it to the same dummy switch for motion detection.  The motion detection works and I get alerts through Homekit.  However, it doesn't appear to start HKSV recording.  The HKSV recording only works with RTSP.   Does anyone have HKSV recording working with an ONVIF camera?

**Comment by ptrangithub** [comment]
I am following your footsteps, however, I noticed that I don't receive any motion detected on the camera.

I made sure to test the email and the Dummy Sensor console shows that it receives the email/message. It's just that the camera itself doesn't generate the event.

Now, when I was initially setting up the Floodlight, it does detect motion and person detection, but after several settings, it (the Floodlight) just doesn't send the event anymore.

Also, when I do a `test email` trigger from the, HKSV doesn't pick it up. If I added the `OpenCV Motion Detection` extension it works, but this is only a test as I want to rely on the Floodlight to detect and forward the event to Scrypted, then do HomeKit.

Do you experience the same thing?

**Reply by jgbaum** [reply]
I never got it working with ONVIF.  I had it working with RTSP, including HKSV recording.  However, there was too much of a lag between when the motion started and the start of the video.  So I ended up using the built-in FTP snapshot upload functionality of the Foscam and have this app (https://github.com/jgbaum/DirHooker) monitor the directory where the snapshots are uploaded for changes.  Once a new snapshot is uploaded, it triggers a webhook to the custom motion sensor in scrypted.  It works surprisingly well for me and there's almost no lag between when the motion starts and the HKSV clips that are recorded.

**Reply by ptrangithub** [reply]
I'm replying to add info and help others and hope that there will be new changes in the future.

1. Like `jgbaum` said, I had to Foscam Floodlight as `rtsp camera`.
2. Using `dummy detection` + `mail plugin` to relay the event from the camera doesn't work well.
  - I think the `dummy switch` does work, however, Foscam Floodlight's motion/human detection is really bad, you literally had to walk really close to it for it to trigger.
3. I end up using `Open CV detection plugin` extension for the Foscam Floodlight. This works well enough for me. It triggers HKSV and recordings correctly, no missing footage. One small quirks is that because it uses Open CV, the notification is a bit slow. As in, compared to `Reolink Doorbell`, I get the event on HomeHub almost instantaneously, where as Foscam/rtsp/openCV, I get the event like 1 second later. However, when I view the HKSV, the footage doesn't have any missing footage.
